File: desktop/windows/LedgerLens/Models/TransactionRow.cs
namespace LedgerLens.Models;

/// <summary>
/// One parsed statement line. Field names and semantics match the macOS app and the
/// web/server model so exports are interchangeable across platforms.
/// </summary>
public sealed class TransactionRow
{
    public string Id { get; init; } = "";

    /// <summary>Populated only for multi-file (batch) conversions.</summary>
    public string? SourceFile { get; set; }

    public string Date { get; init; } = "";
    public string ValueDate { get; init; } = "";
    public string Description { get; init; } = "";
    public string Debit { get; init; } = "";
    public string Credit { get; init; } = "";
    public string Balance { get; init; } = "";

    // Display helpers for the UI (empty money renders as an em dash).
    public string DebitDisplay => string.IsNullOrEmpty(Debit) ? "—" : Debit;
    public string CreditDisplay => string.IsNullOrEmpty(Credit) ? "—" : Credit;
    public string BalanceDisplay => string.IsNullOrEmpty(Balance) ? "—" : Balance;
}

File: desktop/macos/LedgerLensDesktop/Models/TransactionRow.swift
import Foundation

/// One parsed statement line. Field names and semantics match the web/server model
/// (`components/conversion-result-view.js` and the API response) so exports line up with
/// what the hosted product produces.
struct TransactionRow: Codable, Identifiable, Hashable {
    let id: String
    /// Populated only for multi-file (batch) conversions, mirroring the server behavior.
    var sourceFile: String?
    let date: String
    let valueDate: String
    let description: String
    let debit: String
    let credit: String
    let balance: String

    /// Reconciliation result: false when this row's `balance` doesn't equal the previous
    /// balance ± this row's amount (set by `StatementConverter.reconcile`). Rows that can't be
    /// checked (no prior balance, missing amount/balance) stay `true`.
    var reconciled: Bool = true
}
